build/cli-builder.sh

```diff
@@ -15,26 +13,31 @@ while IFS=, read -r git_url build_cmd build_dir; do
 
   git_repo=${git_url##*/}
 
-  cd "${REMOTE_SOURCES_DIR}/${git_repo}/${REMOTE_SOURCES_SUBDIR}"
+  cd "${SCRIPT_DIR}/../external/${git_repo}"
+  echo "=="
 
-  # Source relevant cachito.env for downstream builds
-  cachito_path="${REMOTE_SOURCES_DIR}/${git_repo}"/cachito.env
-  if [[ -f "${cachito_path}" ]]; then
-    source "${cachito_path}"
+  # Set branch using .gitmodules config
+  parent_gitmodules="${SCRIPT_DIR}/../.gitmodules"
+  if [[ -f "${parent_gitmodules}" ]]; then
+    submodule_branch=$(git config --file "${parent_gitmodules}" --get "submodule.${git_repo}.branch" 2>/dev/null || echo "")
+    if [[ -n "${submodule_branch}" ]]; then
+      echo "* Creating branch '${submodule_branch}' for version generation"
+      git checkout -b "${submodule_branch}" 2>/dev/null || echo "* Branch '${submodule_branch}' already exists or cannot be created"
+    fi
   fi
```

Comment:
This block was mostly generated by AI, but it took a lot of coaxing to get it to modify the script file. I was a little surprised it started by modifying the git submodules directly and I had to redirect it here. I would have thought it would know better than to modify a submodule.